frameworks/Swift/vapor/Sources/vapor-tfb/main.swift

```diff
@@ -0,0 +1,24 @@
+import Vapor
+
+// Services
+var services = Services.default()
+
+// Router
+let router = EngineRouter.default()
+// Routes
+let json = Message(message: "Hello, world!")
+router.get("json") { req in
+    return json.encode(status: .ok, headers: ["Server": "Vapor"], for: req)
```

Comment:
One of the requirements of the JSON test is:
> For each request, an object mapping the key `message` to `Hello, World!` must be instantiated.

Your implementation looks like it's reusing the same object across all requests.  But I see that it's a `struct` type, so maybe that's not an accurate way to describe what's happening.  Help me understand - What is the difference between (a) what you have now and (b) declaring the `json` object inside of the request handler?  For (b) I'm imagining code like this:

```swift
router.get("json") { req in
    let json = Message(message: "Hello, world!")
    return json.encode(status: .ok, headers: ["Server": "Vapor"], for: req)
```

Is there any difference between those two in terms of performance?

Reply:
No. A struct will need to be copied either way. The performance hit we're taking here is initializing a new `HTTPResponse` (which json.encode) does. If the only requirement for this route is to initialize a new message => hello, world struct then we could make this a lot more efficient by re-using the response struct and just setting the content inside the route closure.

Reply:
Ok.  @bre7 could you please change this to declare the `json` struct inside the route closure?  The way it is now, it looks like exactly what that requirement I cited is there to prevent, and I'd like to get rid of that potential distraction for us as maintainers.  If you want to also implement @tanner0101's suggestion to reuse the `HTTPResponse` struct and set the content inside the route closure too, that's fine with me.